src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -183,66 +176,58 @@ function addScriptTagsToDocument(resourceLoader: WebAssemblyResourceLoader) {
     .keys(resourceLoader.bootConfig.resources.runtime)
     .filter(n => n.startsWith('dotnet.') && n.endsWith('.js'))[0];
   const dotnetJsContentHash = resourceLoader.bootConfig.resources.runtime[dotnetJsResourceName];
-  const scriptElem = document.createElement('script');
-  scriptElem.src = `_framework/${dotnetJsResourceName}`;
-  scriptElem.defer = true;
-
-  // For consistency with WebAssemblyResourceLoader, we only enforce SRI if caching is allowed
-  if (resourceLoader.bootConfig.cacheBootResources) {
-    scriptElem.integrity = dotnetJsContentHash;
-    scriptElem.crossOrigin = 'anonymous';
-  }
+  let src = `./${dotnetJsResourceName}`;
+
 
   // Allow overriding the URI from which the dotnet.*.js file is loaded
   if (resourceLoader.startOptions.loadBootResource) {
     const resourceType: WebAssemblyBootResourceType = 'dotnetjs';
-    const customSrc = resourceLoader.startOptions.loadBootResource(resourceType, dotnetJsResourceName, scriptElem.src, dotnetJsContentHash);
+    const customSrc = resourceLoader.startOptions.loadBootResource(resourceType, dotnetJsResourceName, src, dotnetJsContentHash);
     if (typeof (customSrc) === 'string') {
-      scriptElem.src = customSrc;
+      src = customSrc;
     } else if (customSrc) {
       // Since we must load this via a <script> tag, it's only valid to supply a URI (and not a Request, say)
       throw new Error(`For a ${resourceType} resource, custom loaders must supply a URI string.`);
     }
   }
 
-  document.body.appendChild(scriptElem);
+  return new Promise((resolve, reject) => {
+    // GOTCHA: remove this callback once we switch to ES6
+    // this is callback we have in CJS version of the runtime
+    globalThis.__onDotnetRuntimeLoaded = (createDotnetRuntime) => {
+      delete globalThis.__onDotnetRuntimeLoaded;
+      resolve(createDotnetRuntime);
+    };
+    // this tells webpack to do dynamic import at runtime, not at compile time
+    import(/* webpackIgnore: true */ src).then(({ default: createDotnetRuntime }) => {
```

Comment:
As with the other suggestion above, would it be possible to make this use `await` instead of `then`? It just make the code easier for us to maintain in the future, e.g., if someone comes to add in a `try`/`catch` around it.

Reply:
will do